src/v1/users/controllers/userController.js

```diff
@@ -0,0 +1,90 @@
+import userDao from '../dao';
+import { modelService } from '../services/';
+
+class UserController {
+  constructor(DAO) {
+    this.DAO = DAO;
+    this.create = this.create.bind(this);
+    this.readAll = this.readAll.bind(this);
+    this.readById = this.readById.bind(this);
+    this.readByName = this.readByName.bind(this);
+    this.updateById = this.updateById.bind(this);
+    this.deleteById = this.deleteById.bind(this);
```

Comment:
It doesn't work without the bind?
If no, then can you think of a way to bind them at the method call in genericController (line 10)?

Reply:
Oh, I thought about base class for controllers with this binding, but I decided, that it's redundant. Your advice more interesting.

Reply:
Hmm, it's more difficult than I thought. I can't bind context in genericController, because I get here only method, without controller what should be context for method.

I can do something like service, what will bind all methods in constructor or I can rewrite generic controller, for getting separate controller and method for it.

Something like
```
genericController(permissions.USER, userController, 'readById');
```

Reply:
That's exactly what I thought it should be, good job.
Only thing - do not pass the method name as a string, pass the method itself (with string you don't get intellisense and it's much harder to debug typos).

So it's gonna be 
```
genericController(permissions.USER, userController, userController.readById);
```

Later we can simplify this even further

Reply:
Cool, will do :)